### PR TITLE
fixes slack_configs generation

### PR DESCRIPTION
**Pull request by f41gh7**
for VMAlertmanagerConfig
https://github.com/VictoriaMetrics/operator/issues/339

### CODE REVIEW

File: controllers/factory/alertmanager/config.go [hunk 1]
```diff
@@ -320,7 +320,85 @@ func (cb *configBuilder) buildSlack(slack operatorv1beta1.SlackConfig) error {
...
+	toYaml("callback_id", slack.CallbackID)
+	if slack.LinkNames {
+		temp = append(temp, yaml.MapItem{Key: "link_names", Value: slack.LinkNames})
+	}
+	if slack.ShortFields {
+		temp = append(temp, yaml.MapItem{Key: "short_fields", Value: slack.ShortFields})
+	}
+	if len(slack.MrkdwnIn) > 0 {
+		temp = append(temp, yaml.MapItem{Key: "mrkdwn_in", Value: slack.MrkdwnIn})
+	}
```
Comment (k1rk) [inline controllers/factory/alertmanager/config.go:350]: maybe?
```suggestion
    toYaml("mrkdwn_in", slack.MrkdwnIn)
```

File: controllers/factory/alertmanager/config.go [hunk 2]
```diff
@@ -320,7 +320,85 @@ func (cb *configBuilder) buildSlack(slack operatorv1beta1.SlackConfig) error {
...
+			toConfirm("ok_text", action.ConfirmField.OkText)
+			toConfirm("dismiss_text", action.ConfirmField.DismissText)
+			toConfirm("title", action.ConfirmField.Title)
+			actionYAML = append(actionYAML, yaml.MapItem{Key: "confirm", Value: confirmYAML})
+		}
+		actions = append(actions, actionYAML)
+	}
+	if len(actions) > 0 {
+		temp = append(temp, yaml.MapItem{Key: "actions", Value: actions})
+	}
```
Comment (k1rk) [inline controllers/factory/alertmanager/config.go:383]: same here. makes sense to use same `toYaml`

File: controllers/factory/alertmanager/config.go [hunk 3]
```diff
@@ -320,7 +320,85 @@ func (cb *configBuilder) buildSlack(slack operatorv1beta1.SlackConfig) error {
...
+		}
 
+		if field.Short != nil {
+			fieldYAML = append(fieldYAML, yaml.MapItem{Key: "short", Value: *field.Short})
+		}
+		fields = append(fields, fieldYAML)
+	}
+	if len(fields) > 0 {
+		temp = append(temp, yaml.MapItem{Key: "fields", Value: fields})
+	}
```
Comment (k1rk) [inline controllers/factory/alertmanager/config.go:401]: and here

File: controllers/factory/alertmanager/config.go (hunk 1 above)
Comment (f41gh7) [reply controllers/factory/alertmanager/config.go:350]: toYaml accepts string, its ugly manual yaml building (

Comment (k1rk) [reply controllers/factory/alertmanager/config.go:350]: 👍

**Review by k1rk — APPROVED** [review]

**Pull request merged**